developer/customization/articles/03-architecture/01-architecture-intro.markdown

```diff
@@ -0,0 +1,149 @@
+---
+header-id: architecture
+---
+
+# Architecture
+
+[TOC levels=1-4]
+
+@product@'s architecture helps you build sites that offer powerful functionality
+and terrific styling, by providing these benefits and more:
+
+- Supports using the best, most popular development technologies
+
+- Leverages development standards
+
+- Facilitates swapping implementations
+
+- Starts up fast and performs well
+
+- Its runtime is easy to inspect and configure
+
+The architecture comprises these components: 
+
+**UIs:** Widgets, page content, navigation, admin applications, and themes. 
+
+**Services:** Functionality exposed via Java APIs and web APIs. 
+
+**Core:** Web application that bootstraps @product@ and provides a runtime 
+framework for deploying components and customizations. 
+
+@product@ and Liferay apps provide the UI and services that users and external
+components interact with. The UIs and services are the parts you can customize
+in @product@ and add to it. Liferay services do the "heavy lifting". 
+
```

Comment:
TMI. In place of the paragraph above should be a nice diagram fitting the parts together, and then you can illustrate the diagram. Be sure to include project logos in the diagram (OSGi and Spring in the core; services we use---such as Apache CXF or OpenSAML---for Services, and PortletMVC4Spring, JSF, React, Vue, and Angular in the UIs.

Reply:
See our previous efforts at this: 
https://portal.liferay.dev/docs/7-1/tutorials/-/knowledge_base/t/fundamentals#technologies

https://portal.liferay.dev/docs/7-1/tutorials/-/knowledge_base/t/fundamentals#architecture